### ISSUE TITLE
MinEstimator is a terrible default

### ISSUE THREAD

**Issue by AndreasPK** [composewell/bench-show#7]
By default report uses MinEstimator for comparisons as far as I can tell.

```
    | MinEstimator    -- ^ Use 'Mean', 'Median' and 'Regression' estimators for
                      -- both baseline and candidate, and report the estimator
                      -- that shows the minimum difference. This is more robust
                      -- against random variations.
```

Under the current scheme both `[10,10,10,10,10,10]` and `[1,1,1,1,1,55]` compare as equal since the averages are the same.  
In the statistical sense the median is already the most robust metric of the three, dynamically switching between them can only make robustness worse.

I suggest using the Median by default instead.

**Comment by harendra-kumar** [comment]
I did a lot of experiments on this. My goal/motivation was to automatically point out which benchmarks have regressed or improved. The primary testbed/motivation for this was the "streamly" package. I ran the streamly benchmarks repeatedly on my MacBook without changing the code at all. With a single estimator I could not get satisfactory results, they always varied quite a bit, more than what I desired, but `minEstimator` could get the results within 3-5% run to run variation.

The default was kept keeping that use in mind. I am open to changing the default to Median though. I agree, median is quite robust in general. Median may be pedantically correct, but minEstimator is perhaps what most people would want in practice, that's why I kept that as default. If we are not among the majority we have the option to switch to another estimator. 

I agree this is terrible in theory, we cannot reason about it well enough, but in practice it works quite well, perhaps because we do not usually see outliers like you gave in the example above. If we have such outliers in the results, perhaps there is something wrong with the measurements and they may not be usable in the first place, using any estimator. Maybe we can detect this automatically and point that out as a warning.  BTW, in verbose mode it reports which method was chosen for each benchmark and standard deviation/r-square as well. Maybe we can mark the ones with high deviation in red color and report the deviation as well.
